src/main/java/reposense/parser/CsvParser.java

```diff
@@ -56,7 +56,10 @@
     private int numOfLinesBeforeFirstRecord = 0;
 
     /**
+     * Creates {@code CsvParser} with given {@code csvFilepath}.
+     *
      * @throws IOException if {@code csvFilePath} is an invalid path.
+     * @throws FileNotFoundException if the csv file cannot be found in the provided {@code csvFilePath}.
      */
     public CsvParser(Path csvFilePath) throws IOException {
```

Comment:
I think we should add a `throws FileNotFoundException` into the method signature.